chips/cc26x2/src/rfc.rs

```diff
@@ -0,0 +1,580 @@
+    }
+
+    fn wait_cmdr(&self, rf_command: u32) -> RfcResult {
+        let command_op: &cmd::CmdCommon = unsafe { &*(rf_command as *const cmd::CmdCommon) };
+
+        let mut timeout: u32 = 0;
+        let mut status = 0;
+        const MAX_TIMEOUT: u32 = 0x2FFFFFF;
+        while timeout < MAX_TIMEOUT {
+            status = command_op.status.get();
+            if status == 0x0400 {
+                return Ok(());
+            }
+
+            timeout += 1;
+        }
+
+        return Err(status as u32);
+    }
+
+    fn cmdsta(&self) -> u32 {
+        let dbell_regs = RFC_DBELL_BASE;
+        let ret: u32 = dbell_regs.cmdsta.get();
+
+        return ret;
+    }
+
+    fn enable_hw_interrupts(&self) {
+        let dbell_regs = RFC_DBELL_BASE;
+        // Enable all interrupts
+        dbell_regs
+            ._rfhwien
+            .modify(RFHWInterrupts::ALL_INTERRUPTS::SET);
+    }
+
+    pub fn handle_hw_interrupts(&self) {
+        let dbell_regs = RFC_DBELL_BASE;
+        // Clear all RFHW interrupts
+        dbell_regs
+            ._rfhwifg
+            .write(RFHWInterrupts::ALL_INTERRUPTS::SET);
+    }
+
+    fn disable_hw_interrupts(&self) {
+        let dbell_regs = RFC_DBELL_BASE;
+        // Disable all RFHW interrupts
+        dbell_regs
+            ._rfhwien
+            .modify(RFHWInterrupts::ALL_INTERRUPTS::CLEAR);
+        // Clear all RFHW interrupts
+        dbell_regs
+            ._rfhwifg
+            .write(RFHWInterrupts::ALL_INTERRUPTS::SET);
+    }
+
+    fn enable_cpe_interrupts(&self) {
+        let dbell_regs = RFC_DBELL_BASE;
+        // Enable CPE interrupts
+        dbell_regs
+            .rfcpeien
+            .modify(CPEInterrupts::ALL_INTERRUPTS::SET);
+    }
+
+    pub fn handle_cpe_interrupts(&self) {
+        let dbell_regs = RFC_DBELL_BASE;
+        // Clear all CPE interrupts
+        dbell_regs.rfcpeifg.set(0x7FFFFFFF);
+    }
+
+    fn disable_cpe_interrupts(&self) {
+        let dbell_regs = RFC_DBELL_BASE;
+        // Disable all CPE interrupts
+        dbell_regs
+            .rfcpeien
+            .modify(CPEInterrupts::ALL_INTERRUPTS::CLEAR);
+        // Clear all CPE interrupts
+        dbell_regs.rfcpeifg.set(0x7FFFFFFF);
+    }
+
+    pub fn cpe_vec_select(&self, cpe: bool) {
+        let dbell_regs = RFC_DBELL_BASE;
+        // Select CPE0 or CPE1
+        if !cpe {
+            dbell_regs.rfcpeisl.modify(CPEVectorSelect::ALL::CLEAR);
+        } else {
+            dbell_regs.rfcpeisl.modify(CPEVectorSelect::ALL::SET);
+        }
+    }
+
+    pub fn handle_ack_interrupt(&self) {
+        let dbell_regs = RFC_DBELL_BASE;
+        // Reset flag for Command ACK
+        dbell_regs.rfackifg.write(DBellCmdAck::CMDACK::SET);
+    }
+
+    pub fn send<T>(&self, rf_command: &T) -> RfcResult {
+        let command = { (rf_command as *const T) as u32 };
+
+        return self.post_cmdr(command);
+    }
+
+    fn send_direct(&self, dir_command: &cmd::DirectCommand) -> RfcResult {
```

Comment:
This is OK, but if `cmd::DirectCommand` is `Copy`, and because it's the size of a register anyway, the argument doesn't need to be a reference (in any case all of this will likely get inlined by the compiler so it shouldn't have an impact on performance/code-size/memory regardless).

Reply:
edited